app/herdbook.py

```diff
@@ -474,6 +478,137 @@ def get_mean_kinship(g_id):
     return {}
 
 
+@APP.route("/api/certificates/issue/<i_number>")
+@login_required
+def issue_certificate(i_number):
+    """
+    Returns an issued pdf certificate of the individual given by `i_number`.
+    """
+    user_id = session.get("user_id", None)
+    ind = da.get_individual(i_number, user_id)
+
+    if ind:
+        all_data = get_all_data(ind, user_id)
+        pdf_bytes = get_certificate(all_data)
+        date = datetime.datetime.utcnow()
+        date = date.strftime("%Y-%m-%d_%H%M%S%f")
+        object_name = f"{date}.pdf"
+        response = make_response(pdf_bytes.getvalue())
+        response.headers["Content-Type"] = "application/pdf"
+        response.headers["Content-Disposition"] = "inline; filename=%s" % object_name
+        return response
+
+    return jsonify({"response": "Individual not found"}, 404)
+
+
+def flatten_list_of_dcts(in_list):
+    """
+    Flattens a list of dictionaries
+    """
+    dct = {}
+    for item in in_list:
+        if item is not None:
+            dct.update(item)
+    return dct
+
+
+def get_all_data(ind, user_id):
+    """
+    Gets all data with regard to an individual inclusing its pedigree.
+    """
+    parent_keys = [
+        (1, "F"),
+        (1, "M"),
+        (2, "F"),
+        (2, "MF"),
+        (2, "FM"),
+        (2, "MM"),
+    ]
+    inds_data_lst = []
+
+    inds_data_lst.append(ind)
+    for level, a_type in parent_keys:
+        inds_data_lst.append(_get_parent(ind, user_id, level, a_type))
+
+    return flatten_list_of_dcts(inds_data_lst)
+
+
+def _get_parent(ind, user_id, ancestry_level, ancestry_type):
+    ancestries = {
+        (1, "F"): "father",
+        (1, "M"): "mother",
+        (2, "F"): "father",
+        (2, "MF"): "father",
+        (2, "FM"): "mother",
+        (2, "MM"): "mother",
+    }
```

Comment:
Is this used elsewhere? Otherwise it's seems to complicate stuff rather than simplifying to me.

Reply:
Complicate what stuff? And no, I don't think it's used elsewhere.

Reply:
I haven't tried rewritring it, but my spontaneous feeling is it would be simpler having it as a single function.